internal/api/middleware/timer.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-kit/kit/metrics"
+)
+
+func getIdentifier(ctx context.Context) string {
+	str := strings.ReplaceAll(chi.RouteContext(ctx).RoutePattern(), "/", ".")
+	regx := regexp.MustCompile(`[{}]`)
+
+	str = regx.ReplaceAllString(str, "")
```

Comment:
We can use a [`strings.Replacer`](https://golang.org/pkg/strings/#Replacer) to handle all of the string substitutions in a single pass, and it's more efficient than regex: https://play.golang.org/p/bZ1J__BLLQR